xds/src/test/java/io/grpc/xds/internal/sds/SslContextProviderSupplierTest.java

```diff
@@ -57,24 +59,26 @@ public void setUp() {
     MockitoAnnotations.initMocks(this);
   }
 
-  private void prepareSupplier() {
+  private void prepareSupplier(boolean callUpdateSslContext) {
     upstreamTlsContext =
             CommonTlsContextTestsUtil.buildUpstreamTlsContextFromFilenames(
                     CLIENT_KEY_FILE, CLIENT_PEM_FILE, CA_PEM_FILE);
     mockSslContextProvider = mock(SslContextProvider.class);
     doReturn(mockSslContextProvider)
             .when(mockTlsContextManager)
             .findOrCreateClientSslContextProvider(eq(upstreamTlsContext));
-    mockCallback = mock(SslContextProvider.Callback.class);
-    Executor mockExecutor = mock(Executor.class);
-    doReturn(mockExecutor).when(mockCallback).getExecutor();
     supplier = new SslContextProviderSupplier(upstreamTlsContext, mockTlsContextManager);
-    supplier.updateSslContext(mockCallback);
+    if (callUpdateSslContext) {
+      mockCallback = mock(SslContextProvider.Callback.class);
+      Executor mockExecutor = mock(Executor.class);
+      doReturn(mockExecutor).when(mockCallback).getExecutor();
+      supplier.updateSslContext(mockCallback);
+    }
   }
 
   @Test
   public void get_updateSecret() {
-    prepareSupplier();
+    prepareSupplier(true);
```

Comment:
I would suggest use two separate methods:
```
prepareSupplier();
updateSslContext();
```

The boolean arg `true` looks less readable.

Reply:
done